python/test/gluon/test_frontend.py

```diff
@@ -23,20 +31,55 @@ def anonymize_ir(ir):
     return TARGET_PAT.sub('ttg.target = "..."', ir)
 
 
+def structurize_target(target: str) -> GPUTarget:
+    """Turns the given target string into a GPUTarget struct."""
+    backend, arch = target.split(":")
+    if backend == "cuda":
+        arch = int(arch.removeprefix("sm"))
+    return GPUTarget(backend, arch, 32)
+
+
+def aot_compile_gluon(fn, target: GPUTarget, constexprs: dict = {}, signature: dict = {}, attrs: dict = {},
+                      **kwargs) -> CompiledKernel:
+    """AOT compiles the given Gluon kernel and returns the compiled kernel."""
+    src = GluonASTSource(fn=fn, constexprs=constexprs, signature=signature, attrs=attrs)
+    backend = triton.compiler.make_backend(target)
+    options = backend.parse_options(kwargs)
+    result = triton.compile(src, target=target, options=options.__dict__)
+    return result
+
+
+def build_constexpr_dict(**kwargs):
+    """Builds the constexpr dict needed for kernel AOT compilation."""
+    return kwargs
+
+
+def build_signature_dict(**kwargs):
+    """Builds the signature dict needed for kernel AOT compilation."""
+    return {k: triton.runtime.jit.mangle_type(v) for k, v in kwargs.items()}
+
+
+def get_divisibility_16_attr_for_args(indices: List[int]):
```

Comment:
Can we factor these out into a helper file, like `_internal_test.py`? These will be useful in other tests as well

Reply:
Done via ff8aaef.